Makefile

```diff
@@ -457,6 +457,7 @@ export rust_common_flags := --edition=2021 \
 			    -Wclippy::needless_continue \
 			    -Wclippy::no_mangle_with_rust_abi \
 			    -Wclippy::dbg_macro
+export rustdoc_common_flags := --document-private-items
```

Comment:
Documenting private items is a bigger change in terms of impact -- we would need to discuss it internally.

In any case, such a change would be independent from the cleanup.

Reply:
I see, thanks! I have now removed the changes regarding `--document-private-items`. 

I also had to remove the 2 links to `RequestDataWrapper` from the changes, or else we would get:

```
warning: public documentation for `Request` links to private item `RequestDataWrapper`
  --> rust/kernel/block/mq/request.rs:49:19
   |
49 | ///   and valid [`RequestDataWrapper<T>`].
   |                   ^^^^^^^^^^^^^^^^^^^^^ this item is private
   |
   = note: this link will resolve properly if you pass `--document-private-items`
```

I guess it's because `RequestDataWrapper` is only `pub(crate)`, not `pub`:

https://github.com/Rust-for-Linux/linux/blob/a335e95914046c6bed45c0d17cabcd483682cf5e/rust/kernel/block/mq/request.rs#L156

Reply:
Yeah, that is part of why documenting private items is a major change, I would say.